### PR TITLE
Add examples demonstrating TRACESWO and flash read/write for stm32f3discovery

### PR DESCRIPTION
**Pull request by roel-gerrits**
Please let me know if I'm doing this right. Haven't done this merge-request thing before :)

### CODE REVIEW

**Comment by karlp** [comment]
so just fiirst off, one PR for each feature please, wayyyy easier to sanely discuss them as standalone pieces.

File: examples/stm32/f3/stm32f3-discovery/flash/Makefile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+##
+## This file is part of the libopencm3 project.
+##
+## Copyright (C) 2009 Uwe Hermann <uwe@hermann-uwe.de>
```
Comment (karlp) [inline examples/stm32/f3/stm32f3-discovery/flash/Makefile]: This is not true.  This is not a copy of another file, and Uwe didnt' write this file in 2009.  Either drop it completely, or use your own name here.  This comment applies throughout this PR, but I won't comment on it again.

File: examples/stm32/f3/stm32f3-discovery/flash/flash.c [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+
+int main(void)
+{
+	uint16_t counter;
+
+	gpio_setup();
+
+    // read value from flash
+    counter = _my_flash_start;
```
Comment (karlp) [inline examples/stm32/f3/stm32f3-discovery/flash/flash.c]: Try running "make stylecheck" on your tree.  You need consistent tabs, and to use C style comments, not C++ comments.

File: examples/stm32/f3/stm32f3-discovery/flash/flash.c [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    // write the counter's binary value to the LED's
+    gpio_set(GPIOE, (counter & 0xff) << 8);
+
+    // increment counter 
+    counter ++;
```
Comment (karlp) [inline examples/stm32/f3/stm32f3-discovery/flash/flash.c]: also, consider whether the comment actually provides value :)

File: examples/stm32/f3/stm32f3-discovery/traceswo/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# README
+
+This is an example which uses the ITM and TPIU to print 
+debug data to the TRACESWO pin (PB3)
+
+It's intended for the ST STM32F3DISCOVERY eval board. 
+Pin PB3 will output strings in UART compatible format at a baudrate of 115200
```
Comment (karlp) [inline examples/stm32/f3/stm32f3-discovery/traceswo/README.md]: I feel you may be misunderstanding the purpose of trace, or the ways of using it.  Manually setting up the trace blocks, and using such a low baud rate is of marginal interest as a "look, I got one more free tx-only uart, on fixed pin" but is very much _not_ an example of "how to do trace"  You normally want to let your debug tool configure this, based on what you actually want to debug, not be some static setup.  Please read all the comments in https://github.com/libopencm3/libopencm3/pull/1054 for more insight, and feel very welcome to stop by the IRC channel if you want to discuss it further.

**Comment by roel-gerrits** [comment]
Thanks Karl, Thanks for your comments, I think they are fair, I'll try and get them fixed this week. Will close it for now. Thanks for the feedback!

**Pull request closed without merging**